test/unit/Input.test.js

```diff
@@ -0,0 +1,305 @@
+});
+
+describe('Input with (default) type=text', () => {
+    describe('Input without setProps() defined', () => {
+        let input;
+        beforeEach(() => {
+            input = mount(<Input value="initial value" />);
+        });
+        test('Input updates value', () => {
+            expect(input.find('input').getNode().value).toEqual(
+                'initial value'
+            );
+
+            input
+                .find('input')
+                .simulate('change', {target: {value: 'new value'}});
+
+            expect(input.find('input').getNode().value).toEqual('new value');
+        });
+        test('Input does not change state if it rerenders', () => {
+            // dash-renderer could rerender a component with it's original
+            // props, if dash-renderer is not aware of prop changes (that happen with setState
+            // instead of setProps)
+            input.setProps({value: 'new value'});
+
+            // expect value prop to not be updated on state, and on the node itself
+            expect(input.state().value).toEqual('initial value');
+            expect(input.find('input').getNode().value).toEqual(
+                'initial value'
+            );
+        });
+    });
+
+    describe('Input with setProps() defined', () => {
+        let mockSetProps, input;
+        beforeEach(() => {
+            mockSetProps = jest.fn();
+
+            input = mount(
+                <Input value="initial value" setProps={mockSetProps} />
+            );
+        });
+        test('Input does not use state if setProps is defined, and debounce is false', () => {
+            expect(input.state()).toBeFalsy();
+        });
+
+        test('Input will call setProps with value updates if provided', () => {
+            input
+                .find('input')
+                .simulate('change', {target: {value: 'new value'}});
+
+            expect(mockSetProps.mock.calls.length).toEqual(1);
+            expect(mockSetProps.mock.calls[0][0]).toEqual({value: 'new value'});
+        });
+
+        test("Input updates it's value on recieving new props", () => {
+            input.setProps({value: 'new value'});
+
+            // expect value prop to not be updated on state, and on the node itself
+            expect(input.find('input').getNode().value).toEqual('new value');
+        });
+    });
+});
+
+describe('Input with type=number', () => {
+    describe('Without setProps(), using this.state', () => {
+        describe('with min and max props', () => {
+            let input;
+            const props = {
+                value: 0,
+                min: 0,
+                max: 2,
+            };
+            beforeEach(() => {
+                input = mount(<Input type="number" {...props} />);
+            });
+            test('Input can not be updated lower than props.min', () => {
+                input
+                    .find('input')
+                    .simulate('change', {target: {value: props.min - 1}});
```

Comment:
As discussed, these tests are incorrect in so far as they are sending a number to the component when an actual onchange event would send a string